### ISSUE TITLE
It looks more as  ES6 -> ES3 than ES6 -> ES5

### ISSUE THREAD

**Issue by medikoo** [esnext/es6-class#1]
If we're after full ES5 compliance, I think class syntax should be compiled to:

``` javascript
function Person(firstName, lastName) {
    this.firstName = firstName;
    this.lastName = lastName;
}

Object.defineProperties(Person.prototype, {
    name: {
        configurable: true,
        enumerable: false,
        get: function () {
            return this.firstName + ' ' + this.lastName;
        }
    },
    toString: {
        configurable: true,
        enumerable: false,
        writable: true,
        value: function () {
            return this.name;
        }
    }
});
```

And in case of extensions e.g. `class PersonExt extends Person`:

``` javascript
var setPrototypeOf = require('some-object-set-prototype-of-polyfill');

function PersonExt (firstName, lastName) {
    this.firstName = firstName;
    this.lastName = lastName;
}
if (setPrototypeOf) setPrototypeOf(Person, PersonExt);

PersonExtension.prototype = Object.create(Person.prototype, {
    someExtProp: {
        configurable: true,
        enumerable: false,
        writable: true,
        value: 'foo'
    }
});
```

This demands `setPrototypeOf` polyfil, but we can make it optional

**Comment by eventualbuddha** [comment]
Are there functional differences between the two? The only thing I see is that the properties would be non-enumerable.

**Comment by medikoo** [comment]
@eventualbuddha this is important difference, and may affect how some algorithms work.

Other difference is that class constructor of the extension, inherits from other constructor, at least I learned it's going to be the case in ES6 to make sure `@@create` is inherited properly.

It's all quite important if you really write code for ES5+ engines, if you write for ES3+ then indeed your version is good enough.

**Comment by eventualbuddha** [comment]
Sure, I would like to get it as close to the actual behavior of ES6 as
possible. What would be helpful to me is to have some concrete behavior
examples that I can test against. Would you mind posting some examples here
or, better yet, submit a pull request with some failing tests?

**Comment by medikoo** [comment]
Best example is, `for in` iteration on instance, it should not iterate over properties of prototype (Same as `for in` on `{}` won't iteratate `hasOwnProperty`, `isPropertyEnumerable` etc.)

If we make prototype properties `enumerable: true` it will make them iterable.

**Comment by eventualbuddha** [comment]
I believe the new implementation from e771ddae13a0d130aa46c4cb609662ebbbe37122 is what you wanted, so I'm closing this. It's been released as part of v0.1.0. Thanks!

**Comment by medikoo** [comment]
Thank you!
